### PR TITLE
Bump Github action versions and use semantic versioning tags

### PR DESCRIPTION
**Pull request by ggravlingen**
#### Reason for change
Pinning against major version in Github actions may introduce unexpected errors to the workflows if dependencies are updated.

#### Description of change
This PR suggests pinning against the semantic version of all Github actions in use. Updates will be implemented in a controlled manner through the dependabot workflow suggested in #246.

#### Steps to Test

**review**:
@Arelle/arelle

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: .github/workflows/build-linux.yml [hunk 1]
```diff
@@ -36,12 +36,12 @@ jobs:
...
         run: rm -rf xule
       - name: Checkout ixbrl-viewer
-        uses: actions/checkout@v3
+        uses: actions/checkout@v3.0.2
         with:
           repository: Workiva/ixbrl-viewer
           path: ixbrl-viewer
       - name: Set up Node JS
-        uses: actions/setup-node@v3
+        uses: actions/checkout@v3.0.2
```
[the remark this answers is not in the document]
Comment (ggravlingen) [reply .github/workflows/build-linux.yml]: Ouch, good catch!

File: .github/workflows/build-linux.yml [hunk 2]
```diff
@@ -15,16 +15,16 @@ jobs:
 
     steps:
       - name: Checkout arelle
-        uses: actions/checkout@v3
+        uses: actions/checkout@v3.0.2
```
[the remark this answers is not in the document]
Comment (ggravlingen) [reply .github/workflows/build-linux.yml:18]: Sorry, I never saw those. I've pushed those updates now.

**Comment by ggravlingen** [comment]
There is one more instance that I haven't changed here. That is adressed in the PR adding tox though.

**Review by austinmatherne-wk — APPROVED** [review]

**Review by sagesmith-wf — APPROVED** [review]

**Pull request merged**